core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingPublisher.java

```diff
@@ -282,7 +285,15 @@ public void error(Throwable error) {
 
             @Override
             public void subscribe(Subscriber<? super ByteBuffer> s) {
-                delegate.subscribe(s);
+                if (subscribeCalled.compareAndSet(false, true)) {
+                    delegate.subscribe(s);
+                } else {
+                    s.onSubscribe(new NoopSubscription(s));
+                    s.onError(NonRetryableException.create(
+                        "A retry was attempted, but AsyncRequestBody.split does not "
+                        + "support retries. Consider using AsyncRequestBody.fromInputStream with an "
+                        + "input stream that supports mark/reset to get retry support."));
```

Comment:
> Consider using AsyncRequestBody.fromInputStream with an input stream that supports mark/reset to get retry support."

Maybe delete the last sentence since it's not applicable here?